src/js/components/productSupplier/ProductSupplierListTable.jsx

```diff
@@ -1,27 +1,59 @@
 import React, { useMemo } from 'react';
 
 import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
 
 import DataTable, { TableCell } from 'components/DataTable';
 import DateCell from 'components/DataTable/DateCell';
 import PreferenceTypeColumn from 'components/productSupplier/PreferenceTypeColumn';
+import useProductSupplierActions from 'hooks/list-pages/productSupplier/useProductSupplierActions';
 import useProductSupplierListTableData from 'hooks/list-pages/productSupplier/useProductSupplierListTableData';
+import ActionDots from 'utils/ActionDots';
+import CustomModal from 'utils/CustomModal';
+import { hasPermissionsToProductSourceActions } from 'utils/permissionUtils';
 import StatusIndicator from 'utils/StatusIndicator';
 import Translate from 'utils/Translate';
 import ListTableTitleWrapper from 'wrappers/ListTableTitleWrapper';
 import ListTableWrapper from 'wrappers/ListTableWrapper';
 
 const ProductSupplierListTable = ({ filterParams }) => {
+  const { currentUser, isAdmin } = useSelector((state) => ({
+    currentUser: state.session.user,
+    isAdmin: state.session.isUserAdmin,
+  }));
+
+  const {
+    isDeleteConfirmationOpened,
+    deleteConfirmationModalButtons,
+    closeDeleteConfirmationModal,
+    getActions,
+    modalLabels,
+  } = useProductSupplierActions();
+
   const columns = useMemo(() => [
     {
       Header: ' ',
       width: 50,
-      fixed: 'left',
       sortable: false,
       style: {
         overflow: 'visible',
         zIndex: 1,
       },
+      fixed: 'left',
+      Cell: (row) => {
+        const clickableActions = hasPermissionsToProductSourceActions(currentUser, isAdmin)
+          ? { actions: getActions(row.original.id) }
+          : {};
```

Comment:
I think this logic should be moved to the `useProductSupplierActions` hook where we initialize the `getActions()` callback